ocaml/xenopsd/xc/service.ml

```diff
@@ -235,6 +235,14 @@ module type DAEMONPIDPATH = sig
   val name : string
 
   val pid_location : Pid.path
+
+  (* To check if a pid is valid, look up the commandline args for the process
+     pid, check that expected args are included. The expected args are set by
+     (expected_cmdline_args domid) for each service.
+     Note that the parameter "domid" here is required as otherwise we can not
```

Comment:
...and the process name...

In some cases, the process name actually includes the domid, so then we don't need to rely on another `domid` parameter. But we _always_ want to match the process name.

Reply:
Yes, the `domid` is a argument to generate the `expected args`, for example, for qemu, its expected arg is just its process name: "qemu-dm-<domid>" as [configured ](https://github.com/gangj/xen-api/blob/private/gangj/CA-382035/ocaml/xenopsd/xc/service.ml#L395)here.

Reply:
Ah, I see your point, comment updated: https://github.com/xapi-project/xen-api/pull/5325/commits/61354b1cf32471d90a1e55f62b181676ed380ac8